### ISSUE TITLE
Enhanced support for social media 

### ISSUE THREAD

**Issue by daydaya** [MohamedBassem/hoarder-app#97]
We currently obtain information from various social media channels quite often and hope to enhance the capability to crawl and display social media information.

Alternatively, a more ambitious request would be to have a capability and display style similar to the Telegram link preview, which seems to always capture metadata from various social media sources effectively.

The cover image obtained from tweets on platform X is often blank. I think if there is no accompanying image, the cover should display the author's avatar or directly show the icon of platform X.

Additionally, there is a request from a Chinese user. In China, there is a social media platform called Weibo, similar to platform X. When I collect links from Weibo, I often fail to retrieve the data that should be available, like this link: https://weibo.com/6182606334/5022022031903257

On Telegram, the information can be correctly retrieved:

<img width="419" alt="截屏2024-04-13 19 41 48" src="https://github.com/MohamedBassem/hoarder-app/assets/78539920/420e45a3-8f6d-4140-9d13-ded2615edfc5">

However, on Weibo, only this information is visible:

<img width="410" alt="截屏2024-04-13 19 43 00" src="https://github.com/MohamedBassem/hoarder-app/assets/78539920/cec6997e-a8da-4ef9-b50f-875225bd263b">

Thank you very much for looking into this request.

**Comment by MohamedBassem** [comment]
Yeah, I agree with you. I think we can do better in terms of the link preview.
